src/components/Application.ts

```diff
@@ -136,12 +140,14 @@ export const ApplicationFunction: ForwardRefRenderFunction<PixiApplication, Appl
 
         if (canvasElement)
         {
-            if (!rootRef.current)
+            let root = roots.get(canvasElement);
```

Comment:
nit: stylistic choice, I prefer to use const whenever possible to protect against accidental re-assignment, resulting in a behaivor change of the original intention.

```ts
const root = roots.get(canvasElement) ?? createRoot(canvasElement, {}, handleInit)
```